pkg/operation/shoot/shoot_test.go

```diff
@@ -86,6 +85,26 @@ var _ = Describe("shoot", func() {
 		})
 	})
 
+	Describe("#CleanupNetworkRulesEnabled", func() {
+		It("should return true when KubeProxy mode in spec and status are different", func() {
```

Comment:
Please also add a test case for which `shoot.Info.Status.KubeProxyMode` is empty.